src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -2086,56 +2085,70 @@ private static string GetStringForPSCommand(PSCommand psCommand)
             return stringBuilder.ToString();
         }
 
-        private void SetExecutionPolicy(ExecutionPolicy desiredExecutionPolicy)
+        private void SetExecutionPolicy()
         {
-            var currentPolicy = ExecutionPolicy.Undefined;
+            // We want to get the list hierarchy of execution policies
+            // Calling the cmdlet is the simplest way to do that
+            IReadOnlyList<PSObject> policies = this.powerShell
+                .AddCommand("Microsoft.PowerShell.Security\\Get-ExecutionPolicy")
+                    .AddParameter("-List")
+                .Invoke();
 
-            // Get the current execution policy so that we don't set it higher than it already is
-            this.powerShell.Commands.AddCommand("Get-ExecutionPolicy");
+            this.powerShell.Commands.Clear();
 
-            var result = this.powerShell.Invoke<ExecutionPolicy>();
-            if (result.Count > 0)
+            // The policies come out in the following order:
+            // - MachinePolicy
+            // - UserPolicy
+            // - Process
+            // - CurrentUser
+            // - LocalMachine
+            // This is the order of precedence we want to follow, skipping the Process scope
+            //
+            // Get-ExecutionPolicy -List emits PSObjects with Scope and ExecutionPolicy note properties
+            // set to expected values, so we must sift through those.
+            ExecutionPolicy policyToSet = ExecutionPolicy.Bypass;
+            for (int i = policies.Count; i >= 0; i--)
             {
-                currentPolicy = result.FirstOrDefault();
-            }
+                PSObject policyObject = policies[i];
 
-            if (desiredExecutionPolicy < currentPolicy ||
-                desiredExecutionPolicy == ExecutionPolicy.Bypass ||
-                currentPolicy == ExecutionPolicy.Undefined)
-            {
-                this.logger.LogTrace(
-                    string.Format(
-                        "Setting execution policy:\r\n    Current = ExecutionPolicy.{0}\r\n    Desired = ExecutionPolicy.{1}",
-                        currentPolicy,
-                        desiredExecutionPolicy));
-
-                this.powerShell.Commands.Clear();
-                this.powerShell
-                    .AddCommand("Set-ExecutionPolicy")
-                    .AddParameter("ExecutionPolicy", desiredExecutionPolicy)
-                    .AddParameter("Scope", ExecutionPolicyScope.Process)
-                    .AddParameter("Force");
-
-                try
+                if ((ExecutionPolicyScope)policyObject.Members["Scope"].Value == ExecutionPolicyScope.Process)
                 {
-                    this.powerShell.Invoke();
+                    continue;
```

Comment:
```suggestion
                    break;
```

The `*Policy` scopes take precedent over `Process` so the rest can be skipped.

Reply:
Do you mean there's no point looking at the policy options because they will already be set?

Reply:
Yeah.  Like if `MachinePolicy` is `AllSigned`, then it doesn't matter what we set the `Process` scope too because PowerShell won't even look.

Reply:
So in this case though, we start PowerShell with the execution policy set (which I assume is the process policy). So then to set it back to the right one, we look back through the execution policy list.

Are you saying that for a policy-set execution policy, `pwsh -executionpolicy bypass` would be started in the policy-set execution policy rather than bypass and so it's pointless to look?

My original intent here was to just skip over the process policy. So in case we've overridden even a policy-set policy, we'll still inherit it.

Reply:
> Are you saying that for a policy-set execution policy, `pwsh -executionpolicy bypass` would be started in the policy-set execution policy rather than bypass and so it's pointless to look?

Yeah.  Any of the policy scopes take precedent over the `Process` scope which is what `-ExecutionPolicy X` sets.

> My original intent here was to just skip over the process policy. So in case we've overridden even a policy-set policy, we'll still inherit it.

You can't override a GPO set execution policy is what I'm getting at.  Setting `Process` to match either of those is effectively a no-op, so the command invocation can be skipped.

Really though it's unlikely to make a noticeable difference in execution time, so if you'd prefer to keep it how it is in the event that that precedence order changes then that's cool too.

Reply:
No, that makes sense to me. Agreed it won't have performance implications, but it's a small change and if anyone looks at the code later and wonders why, it may as well be correct to our intention